e2e/test/scenarios/question-reproductions/reproductions-2.cy.spec.js

```diff
@@ -554,6 +555,52 @@ describe("issue 30610", () => {
   });
 });
 
+describe("issue 35290", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should render column settings when source query is a table joined on itself (metabase#35290)", () => {
+    const questionDetails = {
+      name: "Orders + Orders",
+      query: {
+        "source-table": ORDERS_ID,
+        joins: [
+          {
+            "source-table": ORDERS_ID,
+            condition: [
+              "=",
+              ["field", ORDERS.ID, null],
+              ["field", ORDERS.ID, null],
+            ],
+            alias: "Orders",
+          },
+        ],
+        limit: 5,
+      },
+    };
+
+    createQuestion(questionDetails).then(({ body: { id: questionId } }) => {
+      const questionDetails = {
+        name: "35290",
+        query: {
+          "source-table": `card__${questionId}`,
+        },
+      };
+
+      createQuestion(questionDetails, { visitQuestion: true });
+    });
+
+    cy.findByTestId("viz-settings-button").click();
+    cy.findByTestId("chartsettings-sidebar")
+      // verify panel is shown
+      .should("contain", "Add or remove columns")
+      // verify column name is shown
+      .should("contain", "Created At");
```

Comment:
It'd be good to also assert that the [error/warning icon](https://github.com/metabase/metabase/assets/6830683/b9cb2038-68e2-46eb-be08-5a0150f24f5f) is not displayed

Reply:
[`b422cba` (#44737)](https://github.com/metabase/metabase/pull/44737/commits/b422cba8f71fba23e0e3be046176179598c534ee)